plugins/cpp/parser/src/cppparser.cpp

```diff
@@ -748,8 +748,21 @@ bool CppParser::parseByJson(
 
   //--- Read the compilation commands compile database ---//
 
-  std::vector<clang::tooling::CompileCommand> compileCommands =
+  std::vector<clang::tooling::CompileCommand> tempCompileCommands =
     compDb->getAllCompileCommands();
+  std::vector<clang::tooling::CompileCommand> compileCommands;
+
+  const std::vector<std::string> cppExts{
+    ".c", ".cc", ".cpp", ".cxx", ".o", ".so", ".a"};
```

Comment:
There is a function already which checks whether a file is a cpp file:
https://github.com/Ericsson/CodeCompass/blob/master/plugins/cpp/parser/src/cppparser.cpp#L157-L166